db/common/src/main/java/io/helidon/db/common/AbstractStatement.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.db.common;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import io.helidon.common.context.Context;
+import io.helidon.common.context.Contexts;
+import io.helidon.common.mapper.MapperManager;
+import io.helidon.db.DbInterceptor;
+import io.helidon.db.DbInterceptorContext;
+import io.helidon.db.DbMapperManager;
+import io.helidon.db.DbStatement;
+import io.helidon.db.DbStatementType;
+
+/**
+ * Common statement methods and fields.
+ *
+ * @param <S> type of a subclass
+ * @param <R> the result type of the statement as returned by {@link #execute()}
+ */
+public abstract class AbstractStatement<S extends DbStatement<S, R>, R> implements DbStatement<S, R> {
```

Comment:
This looks like everything in a single bag.
We have Named statements or anonymous statements (with statement string provided).
It would make sense here to have some kind of hierarchy:
- AbstractStatement :: do the execute
  - AbstractAnonymousStatement :: just pass statement as an argument
  - AbstractNamedStatement :: retrieve statement by name